### ISSUE TITLE
Enable paginationLinks() to set the active class on a parent element

### ISSUE THREAD

**Issue by neokoenig** [cfwheels/cfwheels#908]
This would be useful for things like bootstrap 4 which requires the active class to be set on the parent element, i.e the `<li>` in this example

```
<ul class="pagination">
  <li class="page-item"><a class="page-link" href="#">Previous</a></li>
  <li class="page-item"><a class="page-link" href="#">1</a></li>
  <li class="page-item active"><a class="page-link" href="#">2</a></li>
  <li class="page-item"><a class="page-link" href="#">3</a></li>
  <li class="page-item"><a class="page-link" href="#">Next</a></li>
</ul>
```

Currently, this:
```
set(
functionName="paginationLinks", 
prepend="<ul class='pagination'>", 
append="</ul>", 
prependToPage="<li class='page-item'>", 
appendToPage="</li>", 
linkToCurrentPage=true, 
classForCurrent="page-item active", 
class="page-link", 
anchorDivider="<li class='disabled'><a href='##''>...</a></li>", 
encode="attributes");
```

would get `active` attached to the `<a>` element. Might require a new attribute. Quite how we'd parse what comes into `prependToPage` I don't know yet.

**Comment by geirman** [comment]
class is not part of the [`paginationLinks()`](https://api.cfwheels.org/v2.0#controller.paginationlinks) API definition. This only works because all arguments are passed through to the [`linkTo()`](https://api.cfwheels.org/v2.0#controller.linkto) function. We should probably update the paginationLinks() API definition to note this behavior since this would include data, title, and all other `<a>` tag related attributes. 

I took a quick look at the code to see what changes might be needed to fix this. Seems to me [this block](https://github.com/cfwheels/cfwheels/blob/master/wheels/view/links.cfm#L284-L286) would be the one needing to be modified...

```
if (Len(arguments.prependToPage) && arguments.prependOnAnchor) {
	local.start &= arguments.prependToPage;
}
```

However, it's not clear to me how we'd parse the `prependToPage` html. Is there an example where this has been done elsewhere in the framework? 

Also, would we need to add another attribute to the API? If so, what would we call that? `addActiveClassToPrependedParent`?

**Comment by neokoenig** [comment]
Yeah, you've hit it right on the head - prependTo'x' can accept anything, so it's difficult to define an active class there. You could of course solve it via JS, but that's hardly ideal :)

**Comment by zainforbjs** [comment]
@neokoenig I see that the function that you wrote, is setting the node "**classForCurrent**" as "**page-item active**". This is adding the said class, to the anchor tag. 
The anchor tag should have the "page-link" class. So just replacing your "**page-item** active" with "**page-link** active", the same code works and I am getting the correct bootstrap design for current active link.

The div structure with your function:
```
<ul class="pagination">
    <li class="page-item">
        <a class="page-item active" href="/users?page=1">1</a>
    </li>
    <li class="page-item">
        <a class="page-link" href="/users?page=2">2</a>
    </li>
</ul>
```
![image](https://user-images.githubusercontent.com/115699497/223044207-b18138b9-66ca-44e5-9cd0-a108f7844aeb.png)


The div structure with my changes:
```
<ul class="pagination">
    <li class="page-item">
        <a class="page-link active" href="/users?page=1">1</a>
    </li>
    <li class="page-item">
        <a class="page-link" href="/users?page=2">2</a>
    </li>
</ul>
```
![image](https://user-images.githubusercontent.com/115699497/223044268-9b6ec3b4-2941-4392-8821-2a229ba11e07.png)

**Comment by neokoenig** [comment]
@zainforbjs You've misunderstood the issue I think. 
It's about setting an active class of the parent element - i.e, the result would need to be:

```
<ul class="pagination">
    <li class="page-item active">
        <a class="page-link" href="/users?page=1">1</a>
    </li>
    <li class="page-item">
        <a class="page-link" href="/users?page=2">2</a>
    </li>
</ul>
```

**Comment by zainforbjs** [comment]
@neokoenig I see. Sorry for misunderstanding. 
So, should I go ahead with what @geirman suggested (Adding a new attribute) or should I change the paginationLinks function in some way that using your exact "set" function will render the output as you just suggested?

**Comment by zainforbjs** [comment]
@neokoenig I made some changes required to setting the active class to the immediate parent of the current page element.
I added a new boolean attribute that will set the active class to the prependToPage element, if set to true.

I think this should satisfy the requirement that you mentioned in the issue and it is rendering the correct HTML for paginationLinks according to your comment.

Here is the PR. https://github.com/cfwheels/cfwheels/pull/1254
